static/src/javascripts/projects/common/modules/article/space-filler.spec.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+import { spaceFiller } from 'common/modules/article/space-filler';
+import {
+    findSpace as findSpace_,
+    SpaceError,
+} from 'common/modules/spacefinder';
+import raven from 'lib/raven';
+import { noop } from 'lib/noop';
+
+jest.mock('lib/fastdom-promise');
+jest.mock('common/modules/spacefinder', () => ({
+    findSpace: jest.fn(),
+    // eslint-disable-next-line object-shorthand
+    SpaceError: function() {
+        this.name = 'SpaceError';
+    },
+}));
+
+jest.mock('lib/raven', () => ({
+    captureException: jest.fn(),
+}));
+
+const findSpace: JestMockFn<*, *> = (findSpace_: any);
+
+describe('spacefiller', () => {
+    const writeError = new Error('Mock writer exception');
+
+    afterEach(() => {
+        raven.captureException.mockReset();
+        findSpace.mockReset();
+    });
+
+    it('Returns a promise that resolves when the insertion completes', () => {
+        const rules = {};
+        const els = [document.createElement('div')];
+
+        findSpace.mockReturnValue(Promise.resolve(els));
+
+        const insertion = spaceFiller.fillSpace(rules, elements => elements);
+
+        return expect(insertion).resolves.toBe(els);
+    });
+
+    it('Passes a ruleset to the spacefinder', () => {
+        const rules = {};
+        const insertion = spaceFiller.fillSpace(rules, elements => elements);
+
+        return insertion.then(() => {
+            const spaceFinderArgs = findSpace.mock.calls[0];
+
+            expect(spaceFinderArgs[0]).toBe(rules);
+        });
+    });
+
+    it('If it finds a space, it calls the writer', () => {
+        const rules = {};
+        const writer = jest.fn();
+        const els = [document.createElement('p')];
+
+        findSpace.mockReturnValueOnce(Promise.resolve(els));
+
+        return spaceFiller.fillSpace(rules, writer).then(() => {
+            expect(writer).toHaveBeenCalledWith(els);
+        });
+    });
+
+    it('If there are no spaces, it rejects the promise and does not call the writer', () => {
+        const rules = {
```

Comment:
Are these properties required by `rules` to run this test?

Reply:
Yeah, because [`SpaceError`'s constructor](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/spacefinder.js#L237) expects a [`SpacefinderRules`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/spacefinder.js#L28-L52). I will make this clearer.

Reply:
OK I see, so those properties are required by `SpacefinderRules` type definition.